src/main/jbake/content/blog/2017/05/31/hawkular-grafana-ootb.adoc

```diff
@@ -0,0 +1,253 @@
+
+Let's have a look to the docker-compose file:
+
+[source,yaml]
+.docker-compose.yml
+----
+version: '2' # <1>
+services:
+  hawkular:
+    image: "hawkular/hawkular-services:latest" # <2>
+    ports:
+      - "8080:8080"
+      - "8443:8443"
+      - "9990:9990"
+    # volumes: # <3>
+    #   - /tmp/opt/hawkular/server:/opt/data:Z
+    environment:
+      - CASSANDRA_NODES=hawkular-cassandra
+      - ADMIN_TOKEN
+      - HAWKULAR_AGENT_ENABLE=false # <4>
+  hawkular-cassandra: # <5>
+    image: cassandra:3.0.9
+    environment:
+      - CASSANDRA_START_RPC=true
+  grafana: # <6>
+    image: "hawkular/hawkular-grafana-datasource:latest"
+    ports:
+      - "3000:3000"
+  bootstrap:
+    image: "jotak/hawkular-grafana-bootstrap:latest" # <7>
+    environment:
+      - TENANT=put_your_tenant_here # <8>
+      - DASHBOARDS=put_dashboard_file_names_here # <9>
+----
+<1> Docker-compose format v2 brings some networking improvements that proved to be useful here, that's why you'll need docker 1.10.0+ and docker-compose 1.6.0+.
+<2> Using the official Hawkular Services image. Accessible on http://localhost:8080/hawkular/metrics or http://localhost:8080/hawkular/alerts, with credentials _jdoe_/_password_.
+<3> The volume is commented-out, so data is not persisted if you run `docker-compose down`. To persist data, uncomment these lines and eventually set something different than `/tmp/opt/hawkular/server` as the host directory. Make sure this directory exists on your host.
+<4> The Hawkular agent is disabled by default, which means Hawkular is not self-monitored. You can enable it if you want, and use the `hawkular-self` dashboard.
+<5> The Cassandra node. Note that it's not exposed outside of the compose's subnetwork.
+<6> The Grafana server with pre-installed Hawkular plugin, accessible on http://localhost:3000, with credentials _admin_/_admin_
+<7> This image contains a bootstrap script that configures a Grafana datasource with the given tenant (<8>), which is also injected in dashboard templates defined here (<9>). Those dashboards are sent to Grafana through its REST API. Current available dashboard templates are listed below.
+
+If you want to have more than 1 tenant configured, you can add other sections to this compose file (_bootstrap2_, _bootstrap3_, etc. the name doesn't matter),
+with the same docker image and different environment variables. You are limited, however, in the number of tenants per dashboard (a single dashboard cannot use several tenants).
+You need to duplicate dashboards, or create them manually.
+
+== Available dashboard templates
+
+Only a few are available right now, but the list should grow and pull requests are welcome!
+
+Some dashboards may be very specific, dedicated to the demos presented here, while others aim to be generic and reusable. I guess the next step will be to inject custom dashboards through volume in docker-compose without having the rebuild the bootstrap image. Anyway, the current list is:
```

Comment:
Possible, but it would still require us/me to validate the dashboards (merge PR), whereas a volume allows people to put whatever they want without having to share (think about dashboards very specific to an application, it doesn't have to be shared)